ArduCopter/precision_landing.cpp

```diff
@@ -13,13 +13,11 @@ void Copter::init_precland()
 
 void Copter::update_precland()
 {
-    int32_t height_above_ground_cm = current_loc.alt;
-
-    // use range finder altitude if it is valid, otherwise use home alt
-    if (rangefinder_alt_ok()) {
-        height_above_ground_cm = rangefinder_state.alt_cm_glitch_protected;
+    if (!rangefinder_alt_ok() ||
+        rangefinder_state.alt_cm_glitch_protected == 0) {
+        return precland.update(0, false);
```

Comment:
return a void is legal, but tends to hide bugs, I'd prefer this restructured

Reply:
Removed.

I've sometimes wondered about the return-result-from-void-function thing.  In the case that you change a method from void to boolean it could save you bugs (and having to change functions to handle new boolean return)